providers/oidc.go

```diff
@@ -279,27 +287,39 @@ func (p *OIDCProvider) findClaimsFromIDToken(ctx context.Context, idToken *oidc.
 	return claims, nil
 }
 
-func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface{}) []string {
+func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface{}) ([]string, error) {
 	groups := []string{}
 
 	rawGroups, ok := rawClaims[p.GroupsClaim].([]interface{})
 	if rawGroups != nil && ok {
 		for _, rawGroup := range rawGroups {
-			group, ok := rawGroup.(string)
-			if ok {
-				groups = append(groups, group)
-			}
+			formattedGroups := formatGroup(rawGroup)
+			groups = append(groups, formattedGroups...)
 		}
 	}
 
-	return groups
+	return groups, nil
+
+}
+
+func formatGroup(rawGroup interface{}) []string {
```

Comment:
Should this return a `string` instead of a `[]string`? It doesn't look like `[]string` is getting us anything.

Logically, this makes sense as a 1:1 single in, single out design.